discovery_engine_core/core/src/config.rs

```diff
@@ -58,6 +62,20 @@ pub struct InitConfig {
     pub log_file: Option<String>,
 }
 
+// Importing `InitConfig` in `providers` just to provide a `From` implementation
+// seems wrong, so we're implementing `Into` here instead.
+#[allow(clippy::from_over_into)]
+impl Into<ProviderConfig> for InitConfig {
+    fn into(self) -> ProviderConfig {
+        ProviderConfig {
+            api_base_url: self.api_base_url,
+            api_key: self.api_key,
+            news_provider_path: self.news_provider_path,
+            headlines_provider_path: self.headlines_provider_path,
+        }
+    }
+}
```

Comment:
Because I'm not allowed to, since `ProviderConfig` is from the `providers` crate. I tried to explain this in the comment above the `allow`, but I reckon that didn't explain it properly? 🤔

Reply:
i tried to implement it and my rust analyzer is fine with it. it's weird that this fails to compile for you?!

Reply:
Yep you're right, I think I _expected_ this to not work, but never actually verified that assumption. Works fine 👍🏼